components/autocomplete/index.js

```diff
@@ -3,11 +3,12 @@
  */
```

Comment:
I'm not comfortable reviewing the changes in this file, I feel like it could be simplified if we do the generic `onSelect` callback. I believe @aduth created this file originally and would love if you have time to look at these changes (maybe after the change above)

Reply:
I am happy to explain what I've done line by line if it would help?